rsync_tmbackup.sh

```diff
@@ -32,230 +32,232 @@ trap 'fn_terminate_script' SIGINT
+    echo "                        day, keep one backup per day. After 30 days, keep one backup every 7 days."
+    echo "                        After 365 days keep one backup every 30 days."
+    echo " --no-auto-expire       Disable automatically deleting backups when out of space. Instead an error"
+    echo "                        is logged, and the backup is aborted."
+    echo ""
+    echo "For more detailed help, please see the README file:"
+    echo ""
+    echo "https://github.com/laurent22/rsync-time-backup/blob/master/README.md"
 }
 
 fn_parse_date() {
-	# Converts YYYY-MM-DD-HHMMSS to YYYY-MM-DD HH:MM:SS and then to Unix Epoch.
-	case "$OSTYPE" in
-		linux*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
-		cygwin*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
-		netbsd*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
-		darwin8*) yy=`expr ${1:0:4}`
-			mm=`expr ${1:5:2} - 1`
-			dd=`expr ${1:8:2}`
-			hh=`expr ${1:11:2}`
-			mi=`expr ${1:13:2}`
-			ss=`expr ${1:15:2}`
-			# Because under MacOS X Tiger 'date -j' doesn't work, we do this:
-			perl -e 'use Time::Local; print timelocal('$ss','$mi','$hh','$dd','$mm','$yy'),"\n";' ;;
-		darwin*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
-		FreeBSD*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
-	esac
+    # Converts YYYY-MM-DD-HHMMSS to YYYY-MM-DD HH:MM:SS and then to Unix Epoch.
+    case "$OSTYPE" in
+        linux*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
+        cygwin*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
+        netbsd*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
+        darwin8*) yy=`expr ${1:0:4}`
+            mm=`expr ${1:5:2} - 1`
+            dd=`expr ${1:8:2}`
+            hh=`expr ${1:11:2}`
+            mi=`expr ${1:13:2}`
+            ss=`expr ${1:15:2}`
+            # Because under MacOS X Tiger 'date -j' doesn't work, we do this:
+            perl -e 'use Time::Local; print timelocal('${ss}','${mi}','${hh}','${dd}','${mm}','${yy}'),"\n";' ;;
+        darwin*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
+        FreeBSD*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
+    esac
 }
 
 fn_find_backups() {
-	fn_run_cmd "find "$DEST_FOLDER/" -maxdepth 1 -type d -name \"????-??-??-??????\" -prune | sort -r"
+    fn_run_cmd "find \"$DEST_FOLDER/\" -maxdepth 1 -type d -name \"????-??-??-??????\" -prune | sort -r"
 }
 
 fn_expire_backup() {
-	# Double-check that we're on a backup destination to be completely
-	# sure we're deleting the right folder
-	if [ -z "$(fn_find_backup_marker "$(dirname -- "$1")")" ]; then
-		fn_log_error "$1 is not on a backup destination - aborting."
-		exit 1
-	fi
-
-	fn_log_info "Expiring $1"
-	fn_rm_dir "$1"
+    # Double-check that we're on a backup destination to be completely
+    # sure we're deleting the right folder
+    if [[ -z "$(fn_find_backup_marker "$(dirname -- "$1")")" ]]; then
```

Comment:
change unrelated to this commit (tab to spaces)

any why this change "[[" instead of "["? [Shellcheck.net](shellcheck.net) does not give a hint to do this.